### ISSUE TITLE
Change GPU texture addressing mode

### ISSUE THREAD

**Issue by Yaazarai** [YoYoGames/GameMaker-Bugs#5374]
### Is your feature request related to a problem?

Having to manually check outside screen bounds on simple calcs due to texture clamp to border being the default implementation is unnecessary when the GPU has other addressing modes which solve this and other problems: https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSamplerAddressMode.html

We already have gpu_get_texrepeat_ext(sampler_id) which does the same thing, enabling repeat over border clamping.

### Describe the solution you'd like

Add a new function gpu_texture_addressmode(sampler_id, mode).

Modes:
* Repeat
* Mirrored Repeat
* Clamp to Edge
* Mirrored Clamp to Edge
* Clamp to Border (Border Color technically)

NOTE: Clamp to border returns a specified color within a shader when outside the screen.

### Describe alternatives you've considered

N/A.

**Comment by attic-stuff** [comment]
should be noted that the current runtime doesn't and wont ever use vulkan, but i think most of these should be available in the graphics apis that the current time _does_ use. this fr should also get a GMRT tag to make sure these are included in the upcoming new rendering api

**Comment by Yaazarai** [comment]
> should be noted that the current runtime doesn't and wont ever use vulkan, but i think most of these should be available in the graphics apis that the current time _does_ use. this fr should also get a GMRT tag to make sure these are included in the upcoming new rendering api

Right, the Vulkan API was just for reference with what I was familiar with. I figured other API had similar options.
